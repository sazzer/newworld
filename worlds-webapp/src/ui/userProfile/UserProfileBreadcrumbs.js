// @flow

import React from 'react';
import {Breadcrumb} from 'semantic-ui-react'
import {Link} from 'react-router-dom';
import {Interpolate} from 'react-i18next';
import type {User} from "../../users/users";

/** The flow type representing the props for the User Breadcrumbs */
type UserProfileBreadcrumbsProps = {
    user: User
};

/**
 * The breadcrumbs to display for the user profile
 */
export default function UserProfileBreadcrumbs(props: UserProfileBreadcrumbsProps) {
    return (
        <Breadcrumb>
            <Breadcrumb.Section>
                <Link to="/">
                    <Interpolate i18nKey="userProfile.breadcrumbs.home" />
                </Link>
            </Breadcrumb.Section>
            <Breadcrumb.Divider />
            <Breadcrumb.Section active>
                <Interpolate i18nKey="userProfile.breadcrumbs.current" />
                <Link to="/profile">
                    { props.user.displayName }
                </Link>
            </Breadcrumb.Section>
        </Breadcrumb>
    );
}
