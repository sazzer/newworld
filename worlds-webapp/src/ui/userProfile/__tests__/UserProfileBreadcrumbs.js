// @flow

import React from 'react';
import {shallow} from 'enzyme';
import UserProfileBreadcrumbs  from '../UserProfileBreadcrumbs';

/** Create the component to test */
function setup({user}) {
    return {
        element: shallow(<UserProfileBreadcrumbs user={user} />)
    };
}

describe('UserProfileBreadcrumbs', () => {
    it('Renders correctly', () => {
        const user = {
            id: 'id',
            displayName: 'Test User',
            username: 'testuser'
        };

        const { element } = setup({user});

        expect(element).toMatchSnapshot();
    });
});
