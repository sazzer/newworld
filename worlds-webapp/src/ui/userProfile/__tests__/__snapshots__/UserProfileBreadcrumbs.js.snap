// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileBreadcrumbs Renders correctly 1`] = `
<Breadcrumb>
  <BreadcrumbSection>
    <Link
      replace={false}
      to="/"
    >
      <Interpolate
        className=""
        i18nKey="userProfile.breadcrumbs.home"
      />
    </Link>
  </BreadcrumbSection>
  <BreadcrumbDivider />
  <BreadcrumbSection
    active={true}
  >
    <Interpolate
      className=""
      i18nKey="userProfile.breadcrumbs.current"
    />
    <Link
      replace={false}
      to="/profile"
    >
      Test User
    </Link>
  </BreadcrumbSection>
</Breadcrumb>
`;
